### PR TITLE
Refactor the toolkit to use the new Python_Toolkit

### PR DESCRIPTION
**Pull request by epignatelli**
### NOTE: Depends on 
https://github.com/BHoM/Python_Toolkit/pull/2
It will compile if you're not on that branch, but the toolkit will not work.

### Issues addressed by this PR

Closes #1 

### Test files

The test is done by just compiling the application and checking the `AppData/Roaming/BHoM` directory contains a `python-3.7.3-embed-amd64` folder. Then, it can be tested in tandem with this pr https://github.com/BHoM/Keras_Toolkit/pull/14. `LoadImg` is a simple function to test.

### Changelog

- Centralise the python local installation to `AppData/Roaming/BHoM`
- Add PostBuild events to install python at compile time
- Add the possibility to install modules from post build events

### Additional comments

@FraserGreenroyd this pr does not have labels because they are not set for this repo. 
I will raise an issue for that, would it be ok to pick it up?
@MarthaVoul I added you to this pull request just to start familiarising with what is happening on the framework side. I do not expect you to pick up the review, but you can if you want. So, no pressure :)

### CODE REVIEW

**Comment by epignatelli** [comment]
CTRL C + CTRL V
@alelom @FraserGreenroyd all ready for review! Let me know if you need help with that!

File: Numpy_Engine/Convert/ToNumpy.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Numpy_Engine/Convert/ToNumpy.cs]: Copyright now needs to be 2020

File: Numpy_Engine/Numpy_Engine.csproj [hunk 2]
```diff
@@ -53,11 +53,12 @@
...
-    <Reference Include="Python.Included, Version=3.7.9.2, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\BH.UI.Python.Included.3.7.3.9-alpha\lib\netstandard2.0\Python.Included.dll</HintPath>
-    </Reference>
     <Reference Include="Python.Runtime, Version=3.7.4.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\BH.UI.Python.Runtime.3.7.4-alpha\lib\netstandard2.0\Python.Runtime.dll</HintPath>
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\Python_Toolkit\Build\Python.Runtime.dll</HintPath>
+    </Reference>
+    <Reference Include="Python_Engine">
+      <HintPath>..\..\Python_Toolkit\Build\Python_Engine.dll</HintPath>
```
Comment (FraserGreenroyd) [inline Numpy_Engine/Numpy_Engine.csproj:62]: Copy local should be set to false

File: Numpy_Engine/Numpy_Engine.csproj [hunk 3]
```diff
@@ -88,4 +90,8 @@
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <PropertyGroup>
+    <PostBuildEvent>rem Install keras python package at compile time
+$(SolutionDir)/../Python_Toolkit/Python_PostBuild/bin/Debug/Python_Postbuild.exe "numpy"</PostBuildEvent>
```
Comment (FraserGreenroyd) [inline Numpy_Engine/Numpy_Engine.csproj]: This wasn't working for me - but if I removed `Installing keras in python at compile time` then it did. With that in, it had a 9009 error, but removing that would make it build (but Python did not install - more on that in Python_Toolkit).

File: src/Numpy/Numpy.csproj [hunk 4]
```diff
@@ -34,9 +34,16 @@
...
-    <PackageReference Include="BH.UI.Python.Runtime" Version="3.7.4-alpha" />
     <PackageReference Include="Microsoft.CSharp" Version="4.6.0" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Reference Include="Python.Runtime">
+      <HintPath>..\..\..\Python_Toolkit\Build\Python.Runtime.dll</HintPath>
+    </Reference>
+    <Reference Include="Python_Engine">
+      <HintPath>..\..\..\Python_Toolkit\Build\Python_Engine.dll</HintPath>
```
Comment (FraserGreenroyd) [inline src/Numpy/Numpy.csproj:45]: This wasn't working for me - but if I removed `Installing keras in python at compile time` then it did. With that in, it had a 9009 error, but removing that would make it build (but Python did not install - more on that in Python_Toolkit).

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
I could not get this to work in the way described in the PR body, I could not get Python installer using the dependent toolkits, nor test Numpy_Engine itself I'm afraid. See some comments on the code - might be worth going through this together at some point @epignatelli ?

**Review by FraserGreenroyd — APPROVED** [review]
Have re run the steps in the PR body and am happy this is now working following the changes @epignatelli

**Comment by epignatelli** [comment]
Thanks @FraserGreenroyd!

@alelom since this will be pretty foundational, would you mind trying this as well?

**Comment by epignatelli** [comment]
@alelom , on a second thought, I am merging for the moment. Could you still test this anyway, and raise an issue in case it doesn't work? I am aiming for a wider testing audience.

**Comment by alelom** [comment]
> @alelom , on a second thought, I am merging for the moment. Could you still test this anyway, and raise an issue in case it doesn't work? I am aiming for a wider testing audience.

Works fine. See also my comment https://github.com/BHoM/Keras_Toolkit/pull/14#issuecomment-586209512

**Pull request merged**